### PR TITLE
COM-22: Use stricter tsconfig settings to prevent errors similar to t…

### PR DESCRIPTION
**Pull request by bdsl**
…hat fixed in COM PR #86

See https://github.com/thebiggive/components/pull/86 See https://github.com/thebiggive/components/pull/107

### CODE REVIEW

File: src/app/campaign-groups.service.ts [hunk 1]
```diff
@@ -442,7 +442,7 @@ export class CampaignGroupsService {
     );
 
     // For an unknown/invalid beneficiary, show 'other' symbol.
-    return matchingItems[0].icon || faPlus;
```
Comment (bdsl) [inline src/app/campaign-groups.service.ts:445]: this line specifically is quite similar to the one fixed before. Would have thrown a TypeError if matchingItems[0] was undefined.

File: src/app/home/home.component.html [hunk 2]
```diff
@@ -27,7 +27,7 @@
         space-below='4'
         icon-colour='brand-3'
         background-image-url='/assets/images/card-background-gmf.jpg'
-        main-title='Save the Date for Green Match Fund'
+        main-title='Save the Date for Green Match Fund (text edited in donate-frontend)'
```
Comment (alihejazi) [inline src/app/home/home.component.html:30]:
```suggestion
        main-title='Save the Date for Green Match Fund'
```

File: src/app/search.service.ts [hunk 3]
```diff
@@ -108,7 +108,8 @@ export class SearchService {
   }
 
   hasTermFilterApplied(): boolean {
-    return (this.getQueryParams().term?.length > 0);
+    const term = this.getQueryParams().term || '';
+    return (term.length > 0);
```
Comment (alihejazi) [inline src/app/search.service.ts:112]: Not sure this adds much. The previous code was already type-safe.
```suggestion
    return (this.getQueryParams().term?.length > 0);
```

If possible can you please explain your reasoning for this change?

**Review by alihejazi — APPROVED** [review]

File: src/app/search.service.ts (hunk 3 above)
Comment (bdsl) [reply src/app/search.service.ts:112]: The previous version produces an error with the new tsconfig. `length` could be undefined, so we can't use it as a number:

```

Error: src/app/search.service.ts:111:13 - error TS2532: Object is possibly 'undefined'.

111     return (this.getQueryParams().term?.length > 0);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

Comment (alihejazi) [reply src/app/search.service.ts:112]: Ok. Thanks for explaining.

**Pull request merged**